### PR TITLE
Fixes for windows

### PR DESCRIPTION
**Pull request by bd82**

### CODE REVIEW

**Comment by bd82** [comment]
I don't know if this solves a runtime production bug or not.
But I do know that production code threw a bug during tests on windows so that is likely.

File: package.json [hunk 1]
```diff
@@ -43,19 +43,20 @@
     }
   ],
   "dependencies": {
+    "chevrotain": "3.6.1",
+    "fs-extra": "^6.0.1",
```
Comment (MathieuAA) [inline package.json]: Versions should be fixed

File: lib/utils/file_utils.js [hunk 2]
```diff
@@ -17,8 +17,7 @@
  * limitations under the License.
  */
 
-const fs = require('fs');
-const path = require('path');
+const fs = require('fs-extra');
```
Comment (MathieuAA) [inline lib/utils/file_utils.js]: I'd rather use the `fse` name, so as not get confused between fs and fs-extra.

**Comment by bd82** [comment]
> I don't understand. What bug?

```
"before all" hook failed

Error: ENOENT: no such file or directory, mkdir 'toto\titi\tutu'
    at Object.fs.mkdirSync (fs.js:874:3)
    at folderStructure.reduce (lib\utils\file_utils.js:67:10)
    at Array.reduce (<anonymous>)
    at Object.createDirectory (lib\utils\file_utils.js:63:19)
    at Context.before (test\spec\utils\files_utils_test.js:98:19)
```

File: lib/utils/file_utils.js (hunk 2 above)
Comment (bd82) [reply lib/utils/file_utils.js]: See: https://github.com/jprichardson/node-fs-extra#usage

> fs-extra is a drop in replacement for native fs. All methods in fs are attached to fs-extra. All fs methods return promises if the callback isn't passed.
You don't ever need to include the original fs module again:

Comment (bd82) [reply lib/utils/file_utils.js]: we can still use a separate name if you prefer though...

Comment (MathieuAA) [reply lib/utils/file_utils.js]: I've read that, thanks :) That's why I want the `fse` name.

Comment (bd82) [reply lib/utils/file_utils.js]: sure

**Comment by bd82** [comment]
There are three commits here so please **squash**  😄

File: package.json (hunk 1 above)
Comment (bd82) [reply package.json]: fixed

**Comment by MathieuAA** [comment]
Thank you!!

**Comment by bd82** [comment]
No problem 😄

**Pull request merged**